backend/src/apiserver/server/experiment_server.go

```diff
@@ -328,26 +328,23 @@ func (s *ExperimentServer) canAccessExperiment(ctx context.Context, experimentID
 		// Skip authorization if not multi-user mode.
 		return nil
 	}
-
+	namespace := ""
 	if len(experimentID) > 0 {
 		experiment, err := s.resourceManager.GetExperiment(experimentID)
 		if err != nil {
 			return util.Wrap(err, "Failed to authorize with the experiment ID")
 		}
-		if len(resourceAttributes.Namespace) == 0 {
-			if len(experiment.Namespace) == 0 {
-				return util.NewInternalServerError(
-					errors.New("Empty namespace"),
-					"The experiment doesn't have a valid namespace",
-				)
-			}
-			resourceAttributes.Namespace = experiment.Namespace
-		}
+		namespace = experiment.Namespace
 		if len(resourceAttributes.Name) == 0 {
 			resourceAttributes.Name = experiment.Name
 		}
 	}
-
+	if s.resourceManager.IsEmptyNamespace(resourceAttributes.Namespace) {
+		if len(namespace) == 0 {
+			return util.NewInvalidInputError("An experiment cannot have an empty namespace in multi-user mode")
```

Comment:
Does empty namespace means multi-user mode? It's not super clear to me that this code path is only hit with multi-user mode.

Reply:
`canAccessExperiment()` [returns immediately](https://github.com/kubeflow/pipelines/blob/master/backend/src/apiserver/server/experiment_server.go#L326-L330) if it's a single-user mode.